### DISCUSSION TITLE
On demand ISR issues in preview deployments

### DISCUSSION THREAD

**Discussion by arenclissold** [vercel/community#1475] [Help]
### Summary

Hi 👋,

I having issues with on demand revalidation specifically in preview branches. 
I'm not having this issue locally or in production.

Our preview deployments are set up with branch protection and I'm bypassing the protection correctly.

In the example below I am revalidating the base path `'/'`

Request with `x-vercel-protection-bypass` as a header:
{DOMAIN}/api/revalidate?secret={REVALIDATION_TOKEN}&paths[]=/

From what I can see, my function is failing when calling `res.revalidate('/')`
I then get the below error:
```
Error: Failed to revalidate /: Invalid response 401
    at revalidate (/var/task/node_modules/next/dist/server/api-utils/node.js:297:15)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
    at async /var/task/.next/server/pages/api/revalidate.js:58:7
    at async Promise.all (index 2)
    at async revalidate (/var/task/.next/server/pages/api/revalidate.js:66:30)
    at async Object.apiResolver (/var/task/node_modules/next/dist/server/api-utils/node.js:363:9)
    at async NextNodeServer.runApi (/var/task/node_modules/next/dist/server/next-server.js:487:9)
    at async Object.fn (/var/task/node_modules/next/dist/server/next-server.js:749:37)
    at async Router.execute (/var/task/node_modules/next/dist/server/router.js:253:36)
    at async NextNodeServer.run (/var/task/node_modules/next/dist/server/base-server.js:384:29)
```

Again I'd like to stress this is only with preview deployments.

Below is my revalidation function for reference.
```Typescript
const revalidate = async (
  { query }: RevalidationRequest,
  res: RevalidationResponse
) => {
  if (query.secret !== process.env.REVALIDATION_TOKEN) {
    return handleError(res, 401, 'Invalid token')
  }

  const paths = [query['paths[]']].flat()

  if (!paths[0]) {
    return handleError(res, 400, 'Bad request (no paths)')
  }

  const revalidationPromises = paths.map(async (path) => {
    try {
      await res.revalidate(path)
      console.log(`Revalidated: ${path}`)
    } catch (err) {
      console.log(err)
      return path
    }
  })

  const revalidationResult = await Promise.all(revalidationPromises)

  const failedRevalidations = revalidationResult.filter(
    (path) => path !== undefined
  )

  if (failedRevalidations.length) {
    return handleError(
      res,
      500,
      `Failed Revalidations: ${failedRevalidations.join(', ')}`
    )
  }

  return res.json({ success: true })
}
```

**Comment by amyegan** [comment]
Hi, Aren! Have you set a different revalidation token for your preview environment? 🤔  It looks like your function is returning a `401` when the query doesn't match the expected token

**Reply by arenclissold** [reply]
Hi Amy,
I don't think so, if the revalidation token is incorrect, the api responds with:
```json
{"success":false,"error":"Invalid token"}
```

The error in my question is from Vercel function logs, so this error is occurring on the server.

Does the bypass protection header need to be passed into the res.revalidate function options? As this is where I believe the error is occuring
```ts
await res.revalidate(path, { "x-vercel-protection-bypass": BYPASS_TOKEN })
```
